samtranslator/utils/get_service_regions.py

```diff
@@ -0,0 +1,38 @@
+import boto3
+
+
+def get_all_available_regions(service_name):
+    """
+    Wrapper function to get all available regions including ones that are not regional endpoints
+
+    :param service_name: Name of service to get available regions like 's3' etc.
+    :return:Returns a list of endpoint names (e.g., ["us-east-1"]).
+    """
+
+    return boto3._get_default_session().get_available_regions(service_name, allow_non_regional=True)
+
+
+def get_available_serviceable_regions(service_name):
+    """
+    Wrapper function to get available regions excluding ones that are not regional endpoints
+
+    :param service_name: Name of service to get available regions like 's3' etc.
+    :return:Returns a list of endpoint names (e.g., ["us-east-1"]).
+    """
+
+    return boto3._get_default_session().get_available_regions(service_name, allow_non_regional=False)
+
+
+def get_available_nonserviceable_regions(service_name):
+    """
+    Wrapper function to get only non serviceable or AWS opt-in regions.
+    AWS opt-in regions require users to enable them manually in order to use them.
+
+    :param service_name: Name of service to get available regions like 's3' etc.
+    :return:Returns a list of endpoint names (e.g., ["us-east-1"]).
+    """
+
+    all_regions = boto3._get_default_session().get_available_regions(service_name, allow_non_regional=True)
```

Comment:
you could reuse the above helper functions here.

The docstring still needs clarity, maybe you can clarify by saying which regions are opt-in regions vs which regions are not.

Reply:
I think opt-in regions keeps changing, right? initially a region is opt-in only, but then becomes GA as per my knowledge

Reply:
There are a few possible issues here, one of which is that a new region would require that `boto3` has that region listed, or else it will fail to function. What happens if a region is valid but is not returned by this?